libraries/AP_NavEKF3/AP_NavEKF3.cpp

```diff
@@ -815,23 +815,29 @@ void NavEKF3::UpdateFilter(void)
         }
         runCoreSelection = (imuSampleTime_us - lastUnhealthyTime_us) > 1E7;
     }
+
+    //start updating errors for all cores after core selection logic is true
+    if(runCoreSelection) {
+        updateCoreErrors();
+    }
+
     float primaryErrorScore = core[primary].errorScore();
     if ((primaryErrorScore > 1.0f || !core[primary].healthy()) && runCoreSelection) {
```

Comment:
this if statement means we only consider lane changes if the primary is unhealthy or it's score is high. I think we need to consider switching if the accumulated difference between the primary and an alternative lane goes over a threshold

Reply:
I think that logic is this one just below it - 
`if ((primaryErrorScore > 1.0f || !core[primary].healthy()) && runCoreSelection)`
what the new if condition stands for is to start accumulating errors only after core selection logic is available (the if block just above it does that). The reason for waiting till core selection is available is to not accumulate errors that happen due to the initialisation process. Please correct me if I'm wrong. 
I think we should add a separate check that ensures even when the current primary is healthy or is returning an NavEKF_core::errorScore() < 1.0f,  if difference between accumulated error between primary and another lane is high above a threshold, we should switch.

Reply:
sounds good, thanks